### PR TITLE
Add filtering for articles 

### PR DESCRIPTION
**Pull request by kidzegeye**
## Overview

Add article filtering capability based on keywords defined in [volume microservice](https://github.com/cuappdev/volume-microservice).

## Changes Made

Added a new field `filter` and filtered articles on all article queries that had `filter: true`. Will probably need a migration script to modify existing articles on dev/prod.

## Test Coverage

Tested through manual testing by checking whether articles with `filter=true` would appear when queried.

## Related PRs or Issues

https://github.com/cuappdev/volume-microservice/pull/12

### CODE REVIEW

**Review by katesliang — DISMISSED** [review]
LGTM

File: src/repos/ArticleRepo.ts [hunk 1]
```diff
@@ -5,24 +5,34 @@ import { DEFAULT_LIMIT, MAX_NUM_DAYS_OF_TRENDING_ARTICLES } from '../common/cons
 import { PublicationModel } from '../entities/Publication';
 
 const getArticleByID = async (id: string): Promise<Article> => {
-  return ArticleModel.findById(new ObjectId(id));
+  const article = ArticleModel.findById(new ObjectId(id));
+  if((await article).filtered){
+    return article;
+  }
+  return null;
```
Comment (dishanest) [inline src/repos/ArticleRepo.ts:12]: doesn't this return `null` if `filtered=false`?

**Review by dishanest — CHANGES_REQUESTED** [review]
Just one question on `getArticleByID`

File: src/repos/ArticleRepo.ts (hunk 1 above)
Comment (kidzegeye) [reply src/repos/ArticleRepo.ts:12]: good catch lol

**Review by dishanest — APPROVED** [review]
🤩🤑

**Review by snajima — DISMISSED** [review]
Looks good!

File: src/entities/Article.ts [hunk 2]
```diff
@@ -42,6 +42,10 @@ export class Article {
   @Field()
   @Property({ default: false })
   isTrending: boolean;
+
+  @Field()
+  @Property()
+  filtered: boolean;
```
Comment (chalo2000) [inline src/entities/Article.ts:48]: isFiltered

File: src/repos/ArticleRepo.ts [hunk 3]
```diff
@@ -5,24 +5,34 @@ import { DEFAULT_LIMIT, MAX_NUM_DAYS_OF_TRENDING_ARTICLES } from '../common/cons
 import { PublicationModel } from '../entities/Publication';
 
 const getArticleByID = async (id: string): Promise<Article> => {
-  return ArticleModel.findById(new ObjectId(id));
+  const article = ArticleModel.findById(new ObjectId(id));
+  if((await article).filtered){
```
Comment (chalo2000) [inline src/repos/ArticleRepo.ts:9]: do the await on the previous line

**Review by chalo2000 — DISMISSED** [review]
LGTM - make my style changes and then you can merge.

File: src/repos/ArticleRepo.ts (hunk 3 above)
Comment (kidzegeye) [reply src/repos/ArticleRepo.ts:9]: What do you mean by previous line

Comment (kidzegeye) [reply src/repos/ArticleRepo.ts:9]: I changed it to be more consistent with other filters

**Pull request closed without merging**